### PR TITLE
Feat/improve release quality

### PR DESCRIPTION
**Pull request by getlarge**
- Update PDF generator script (built with parcel-bundler v2)

- Update lint rules

- Create validation and rendering test suites

### CODE REVIEW

File: test/render.spec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const { generatePdf } = require('@s1seven/schema-tools-generate-pdf');
+const { readFileSync } = require('fs');
+const { resolve } = require('path');
+const { fromBuffer } = require('pdf2pic');
+
+describe('Render', function () {
+  const testSuitesMap = [
+    {
+      certificateName: `valid_certificate_1`,
+    },
```
Comment (getlarge) [inline test/render.spec.js:12]: Could you add some new fixtures and declare them here, to make our testing more solid ?

File: test/validate.spec.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    //
+    const validateSchema = ajv.compile(schemaToValidate);
+    expect(() => validateSchema()).not.toThrow();
+  });
+
+  const validCertTestSuitesMap = [
+    {
+      certificateName: `valid_certificate_1`,
+    },
+  ];
```
Comment (getlarge) [inline test/validate.spec.js:24]: Same as above

File: test/validate.spec.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      //
+      const errors = await validate(certificate);
+      expect(errors).toBeNull();
+    });
+  });
+
+  const invalidCertTestSuitesMap = [
+    {
+      certificateName: `invalid_certificate_1`,
+      expectedErrors: {
```
Comment (getlarge) [inline test/validate.spec.js:40]: Here we would we need some extra invalid certificates with broken values to ensure our schema is not letting those values passing the validation

**Review by getlarge — COMMENTED** [review]
I added some tests that be can easily extended by adding some new files to validate against (in `fixtures` folder), and declaring them in the test suites.

**Comment by getlarge** [comment]
@stiebitzhofer Also i changed the rules of the main branch, we cannot push to the main branch anymore without a review.
Strict yes, but that might avoid some mistakes later on.

**Review by stiebitzhofer — APPROVED** [review]

File: test/render.spec.js (hunk 1 above)
Comment (stiebitzhofer) [reply test/render.spec.js:12]: Will do.

File: test/validate.spec.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    const schemaToValidate = JSON.parse(readFileSync(schemaPath, 'utf-8'));
+    const ajv = new Ajv({
+      strictSchema: true,
+      strictNumbers: true,
+      strictRequired: true,
+      strictTypes: true,
+      allErrors: true,
+    });
+    addFormats(ajv);
+    ajv.addKeyword('meta:license');
```
Comment (getlarge) [inline test/validate.spec.js:21]: #13  added strict mode and support for meta:license keyword

**Review by stiebitzhofer — APPROVED** [review]

File: test/helpers.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const { readFileSync } = require('fs');
+const { resolve } = require('path');
+
+const getRefSchemaUrl = () => {
+  const schemaPath = resolve(__dirname, '../schema.json');
+  const schema = JSON.parse(readFileSync(schemaPath, 'utf-8'));
+  return schema.$id;
```
Comment (stiebitzhofer) [inline test/helpers.js:7]: The function returns the value of $id of the local schema file which is `https://schemas.en10204.io/en10168-schemas/v0.0.3/schema.json` so it resolves again to the remote schema.json. Wrote a little helper `getSchema`

Comment (getlarge) [reply test/helpers.js]: switched to AJV

**Pull request merged**